File: backend/fingerprint_reader/fingerprints/models/baby_model.py
from django.db import models
from datetime import date


GENDER_CHOICES = {
  'M': 'masculino',
  'F': 'feminino'
  }


class Baby(models.Model):
  code = models.CharField(max_length=14)
  birth_date = models.DateField(default=date.today)
  gender = models.CharField(choices=GENDER_CHOICES, max_length=1)

  created_at = models.DateTimeField(auto_now=True)
  updated_at = models.DateTimeField(auto_now_add=True)


  def __str__(self):
    return self.code

File: digital_reader/lib/models/baby_model.dart
class BabyModel {
  int? id;
  String? code;
  String? birthDate;
  String? gender;
  String? createdAt;
  String? updatedAt;

  BabyModel(
      {this.id,
      this.code,
      this.birthDate,
      this.gender,
      this.createdAt,
      this.updatedAt});

  BabyModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    code = json['code'];
    birthDate = json['birth_date'];
    gender = json['gender'];
    createdAt = json['created_at'];
    updatedAt = json['updated_at'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['code'] = code;
    data['birth_date'] = birthDate;
    data['gender'] = gender;
    data['created_at'] = createdAt;
    data['updated_at'] = updatedAt;
    return data;
  }
}
